track slick

currently at "slick/slick.git#pull/686/head"

@szeiger on 2.11 I get the following. any idea what this is about?

```
[slick] Processing subproject: osgitests
[slick] All Dependencies for subproject osgitests:
[slick]    org.scala-lang:scala-compiler:2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e:scala-tool->default,optional(default)
[slick]    org.scala-lang:scala-library:2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e:scala-tool->default,optional(default)
[slick]    com.typesafe.slick:slick:3.2.0-dbuildxe2ffc32ea14aa8a82771db64d1264f39f7b0e462:test
[slick]    org.scala-lang:scala-library:2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e
[slick]    com.h2database:h2:1.4.187:test
[slick]    ch.qos.logback:logback-classic:1.1.3:test
[slick]    junit:junit-dep:4.10:test
[slick]    com.novocode:junit-interface:0.11:test
[slick]    org.reactivestreams:reactive-streams:1.0.0:test
[slick]    org.ops4j.pax.exam:pax-exam-container-native:4.6.0:test
[slick]    org.ops4j.pax.exam:pax-exam-junit4:4.6.0:test
[slick]    org.ops4j.pax.exam:pax-exam-link-assembly:4.6.0:test
[slick]    org.ops4j.pax.url:pax-url-aether:1.6.0:test
[slick]    org.ops4j.pax.swissbox:pax-swissbox-framework:1.5.1:test
[slick]    org.apache.felix:org.apache.felix.framework:4.6.1:test
[slick] Running build: osgitests
[slick] Testing: osgitests
[slick] [info] Updating {file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.1/project-builds/slick-e2ffc32ea14aa8a82771db64d1264f39f7b0e462/}osgitests...
[slick] [warn] bnd: Split package, multiple jars provide the same package:slick/util
[slick] [warn] Use Import/Export Package directive -split-package:=(merge-first|merge-last|error|first) to get rid of this warning
[slick] [warn] Package found in   [Jar:classes, Jar:macro-classes]
[slick] [warn] Class path         [Jar:classes, Jar:macro-classes, Jar:scala-library-2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e, Jar:slf4j-api-1.7.10, Jar:config-1.2.1, Jar:reactive-streams-1.0.0, Jar:scala-compiler-2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e, Jar:scala-reflect-2.11.8-dbuildx4b3cb89f76c5caa3cc7ada3b5ebb90c5961d2b8e, Jar:scala-xml-1.0.6-dbuildxcacf386c23eb6cc5d5d2573aa1929fa2b75c0206, Jar:scala-parser-combinators-1.1.0-dbuildxdea5f979a334d6566b075c99de2307b73f0f71f8]
[slick] [warn] bnd: Unused Export-Package instructions: [scala.slick.*] 
[slick] [info] Done updating.
[slick] [info] Compiling 2 Scala sources to /home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.1/project-builds/slick-e2ffc32ea14aa8a82771db64d1264f39f7b0e462/osgi-tests/target/scala-2.11/test-classes...
[slick] [error] Uncaught exception when running slick.osgi.test.BasicTest: java.lang.UnsupportedClassVersionError: org/ops4j/pax/exam/junit/PaxExam : Unsupported major.minor version 51.0
[slick] sbt.ForkMain$ForkError: java.lang.UnsupportedClassVersionError: org/ops4j/pax/exam/junit/PaxExam : Unsupported major.minor version 51.0
[slick]     at java.lang.ClassLoader.defineClass1(Native Method)
[slick]     at java.lang.ClassLoader.defineClassCond(ClassLoader.java:631)
[slick]     at java.lang.ClassLoader.defineClass(ClassLoader.java:615)
[slick]     at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
[slick]     at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
[slick]     at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
[slick]     at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
[slick]     at java.security.AccessController.doPrivileged(Native Method)
[slick]     at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
[slick]     at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
[slick]     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
[slick]     at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
[slick]     at java.lang.Class.forName0(Native Method)
[slick]     at java.lang.Class.forName(Class.java:249)
[slick]     at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:95)
[slick]     at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:107)
[slick]     at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:31)
[slick]     at sun.reflect.annotation.AnnotationParser.parseSig(AnnotationParser.java:370)
[slick]     at sun.reflect.annotation.AnnotationParser.parseClassValue(AnnotationParser.java:351)
[slick]     at sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:280)
[slick]     at sun.reflect.annotation.AnnotationParser.parseAnnotation(AnnotationParser.java:222)
[slick]     at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:69)
[slick]     at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:52)
[slick]     at java.lang.Class.initAnnotationsIfNecessary(Class.java:3079)
[slick]     at java.lang.Class.getAnnotation(Class.java:3038)
[slick]     at com.novocode.junit.JUnitRunner.shouldRun(JUnitRunner.java:169)
[slick]     at com.novocode.junit.JUnitRunner.access$400(JUnitRunner.java:20)
[slick]     at com.novocode.junit.JUnitRunner$1.execute(JUnitRunner.java:123)
[slick]     at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[slick]     at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[slick]     at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[slick]     at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[slick]     at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
[slick]     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
[slick]     at java.lang.Thread.run(Thread.java:662)
[slick] [error] Error: Total 1, Failed 0, Errors 1, Passed 0
[slick] [error] Error during tests:
[slick] [error]     slick.osgi.test.BasicTest
[slick] [error] (osgitests/test:test) sbt.TestsFailedException: Tests unsuccessful
[slick:error] Incomplete(node=Some(ScopedKey(Scope(Select(ProjectRef(file:/home/jenkins/workspace/scala-2.11.x-integrate-community-build/target-0.9.1/project-builds/slick-e2ffc32ea14aa8a82771db64d1264f39f7b0e462/,osgitests)),Select(ConfigKey(test)),Global,Global),test)), tpe=Error, msg=None, causes=List(), directCause=Some(sbt.TestsFailedException: Tests unsuccessful))
[slick:error]   at sbt.Incomplete.copy(Incomplete.scala:17)
[slick:error]   at sbt.EvaluateTask$$anonfun$taskToKey$1.apply(EvaluateTask.scala:400)
[slick:error]   at sbt.EvaluateTask$$anonfun$taskToKey$1.apply(EvaluateTask.scala:399)
[slick:error]   at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
[slick:error]   at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
[slick:error]   at sbt.Incomplete$$anonfun$sbt$Incomplete$$visit$1$1.apply(Incomplete.scala:32)
[slick:error]   at sbt.Incomplete$$anonfun$sbt$Incomplete$$visit$1$1.apply(Incomplete.scala:32)
[slick:error]   at scala.collection.mutable.MapLike$class.getOrElseUpdate(MapLike.scala:189)
[slick:error]   at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:91)
[slick:error]   at sbt.Incomplete$.sbt$Incomplete$$visit$1(Incomplete.scala:32)
[slick:error]   at sbt.Incomplete$.transform(Incomplete.scala:36)
[slick:error]   at sbt.Incomplete$.transformBU(Incomplete.scala:26)
[slick:error]   at sbt.EvaluateTask$$anonfun$transformInc$1.apply(EvaluateTask.scala:398)
[slick:error]   at sbt.EvaluateTask$$anonfun$transformInc$1.apply(EvaluateTask.scala:398)
[slick:error]   at scala.util.Either$LeftProjection.map(Either.scala:377)
[slick:error]   at sbt.EvaluateTask$.transformInc(EvaluateTask.scala:398)
[slick:error]   at sbt.EvaluateTask$.run$1(EvaluateTask.scala:364)
[slick:error]   at sbt.EvaluateTask$.runTask(EvaluateTask.scala:378)
[slick:error]   at sbt.EvaluateTask$$anonfun$apply$1$$anonfun$apply$3.apply(EvaluateTask.scala:279)
[slick:error]   at sbt.EvaluateTask$$anonfun$apply$1$$anonfun$apply$3.apply(EvaluateTask.scala:279)
[slick:error]   at scala.Option$WithFilter.map(Option.scala:206)
[slick:error]   at sbt.EvaluateTask$$anonfun$apply$1.apply(EvaluateTask.scala:279)
[slick:error]   at sbt.EvaluateTask$$anonfun$apply$1.apply(EvaluateTask.scala:278)
[slick:error]   at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:314)
[slick:error]   at sbt.EvaluateTask$.apply(EvaluateTask.scala:278)
[slick:error]   at sbt.EvaluateTask$.apply(EvaluateTask.scala:276)
[slick:error]   at sbt.Extracted.runTask(Extracted.scala:44)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$.com$typesafe$dbuild$DistributedRunner$$buildTestPublish$1(DistributedRunner.scala:596)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$19.apply(DistributedRunner.scala:616)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$19.apply(DistributedRunner.scala:616)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$com$typesafe$dbuild$DistributedRunner$$runAggregate$1.apply(DistributedRunner.scala:70)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$com$typesafe$dbuild$DistributedRunner$$runAggregate$1.apply(DistributedRunner.scala:67)
[slick:error]   at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:111)
[slick:error]   at scala.collection.immutable.List.foldLeft(List.scala:84)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$.com$typesafe$dbuild$DistributedRunner$$runAggregate(DistributedRunner.scala:67)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$14.apply(DistributedRunner.scala:574)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$14.apply(DistributedRunner.scala:574)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$.buildStuff(DistributedRunner.scala:616)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$generateArtifacts$1.apply(DistributedRunner.scala:745)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$generateArtifacts$1.apply(DistributedRunner.scala:745)
[slick:error]   at com.typesafe.dbuild.StateHelpers$.saveLastMsg(StateHelpers.scala:35)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$.generateArtifacts(DistributedRunner.scala:745)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$com$typesafe$dbuild$DistributedRunner$$buildIt$1.apply(DistributedRunner.scala:749)
[slick:error]   at com.typesafe.dbuild.DistributedRunner$$anonfun$com$typesafe$dbuild$DistributedRunner$$buildIt$1.apply(DistributedRunner.scala:749)
[slick:error]   at sbt.Command$$anonfun$command$1$$anonfun$apply$1.apply(Command.scala:30)
[slick:error]   at sbt.Command$$anonfun$command$1$$anonfun$apply$1.apply(Command.scala:30)
[slick:error]   at sbt.Command$.process(Command.scala:93)
[slick:error]   at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:98)
[slick:error]   at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:98)
[slick:error]   at sbt.State$$anon$1.process(State.scala:184)
[slick:error]   at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:98)
[slick:error]   at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:98)
[slick:error]   at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[slick:error]   at sbt.MainLoop$.next(MainLoop.scala:98)
[slick:error]   at sbt.MainLoop$.run(MainLoop.scala:91)
[slick:error]   at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:70)
[slick:error]   at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:65)
[slick:error]   at sbt.Using.apply(Using.scala:24)
[slick:error]   at sbt.MainLoop$.runWithNewLog(MainLoop.scala:65)
[slick:error]   at sbt.MainLoop$.runAndClearLast(MainLoop.scala:48)
[slick:error]   at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:32)
[slick:error]   at sbt.MainLoop$.runLogged(MainLoop.scala:24)
[slick:error]   at sbt.StandardMain$.runManaged(Main.scala:53)
[slick:error]   at sbt.xMain.run(Main.scala:28)
[slick:error]   at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:57)
[slick:error]   at xsbt.boot.Launch$.withContextLoader(Launch.scala:77)
[slick:error]   at xsbt.boot.Launch$.run(Launch.scala:57)
[slick:error]   at xsbt.boot.Launch$$anonfun$explicit$1.apply(Launch.scala:45)
[slick:error]   at xsbt.boot.Launch$.launch(Launch.scala:65)
[slick:error]   at xsbt.boot.Launch$.apply(Launch.scala:16)
[slick:error]   at xsbt.boot.Boot$.runImpl(Boot.scala:32)
[slick:error]   at xsbt.boot.Boot$.main(Boot.scala:21)
[slick:error]   at xsbt.boot.Boot.main(Boot.scala)
[slick:error] Caused by: sbt.TestsFailedException: Tests unsuccessful
[slick:error]   at sbt.TestResultLogger$Defaults$Main.run(TestResultLogger.scala:95)
[slick:error]   at sbt.Defaults$$anonfun$testTasks$5.apply(Defaults.scala:421)
[slick:error]   at sbt.Defaults$$anonfun$testTasks$5.apply(Defaults.scala:418)
[slick:error]   at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[slick:error]   at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
[slick:error]   at sbt.std.Transform$$anon$4.work(System.scala:63)
[slick:error]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[slick:error]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[slick:error]   at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[slick:error]   at sbt.Execute.work(Execute.scala:235)
[slick:error]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[slick:error]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[slick:error]   at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[slick:error]   at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[slick:error]   at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[slick:error]   at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[slick:error]   at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
[slick:error]   at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[slick:error]   at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[slick:error]   at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
[slick:error]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
[slick:error]   at java.lang.Thread.run(Thread.java:662)
[slick:error] java.lang.RuntimeException: Tests unsuccessful
[slick:error]   at scala.sys.package$.error(package.scala:27)
[slick:error]   at distributed.support.sbt.SbtRunner.distributed$support$sbt$SbtRunner$$processCommand(SbtRunner.scala:93)
[slick:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:74)
[slick:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:74)
[slick:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:74)
[slick:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:64)
[slick:error]   at sbt.IO$.withTemporaryFile(IO.scala:311)
[slick:error]   at distributed.support.sbt.SbtRunner.run(SbtRunner.scala:64)
[slick:error]   at distributed.support.sbt.SbtBuilder$.buildSbtProject(SbtBuildRunner.scala:84)
[slick:error]   at distributed.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:68)
[slick:error]   at distributed.support.sbt.SbtBuildSystem.runBuild(SbtBuildSystem.scala:18)
```

this is from https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-community-build/85/console

looks like the same thing when building on JDK8 too: https://scala-ci.typesafe.com/job/scala-2.11.x-jdk8-integrate-community-build/54/consoleFull

--- a/common.conf
+++ b/common.conf
@@ -77,8 +77,6 @@ vars: {
   parboiled-ref                : "sirthias/parboiled.git#c53e650212f222c9b1f75fa1ab13d7cab9db164e"
   // This pull request supports the fully-cross-versioned continuations plugin, shipped with 2.11+
   scala-arm-ref                : "jsuereth/scala-arm.git#pull/31/head"
-  // fixed sha/tag (a compromise)
-  slick-ref                    : "slick/slick.git#pull/686/head"
   genjavadoc-ref               : "typesafehub/genjavadoc.git#v0.5_2.11.0-M8"
   // There's no suitable release yet, so we depend on this commit. Upgrade to the next tagged version once released
   scala-refactoring-ref        : "scala-ide/scala-refactoring.git#44dca8b74808528693f884cfd3c5c9d3ed13e519"
@@ -106,6 +104,7 @@ vars: {
   pimpathon-ref                : "stacycurl/pimpathon.git#d2354dd92f5481610f4610edba3574880b07263e"
   scalariform-ref              : "adriaanm/scalariform.git#community-build"
   scalamock-ref                : "adriaanm/scalamock.git#community-build"
+  sbt-testng-interface-ref     : "SethTisue/sbt-testng-interface.git#no-bintray"
 
   // tracking upstream (the ideal)
   akka-ref                     : "akka/akka.git#release-2.3"
@@ -127,6 +126,7 @@ vars: {
   scalaz-ref                   : "scalaz/scalaz.git#series/7.1.x"
   discipline-ref               : "typelevel/discipline.git#v0.2"
   scala-js-ref                 : "scala-js/scala-js.git"
+  slick-ref                    : "slick/slick.git#3.1"   // stay here; master wants Java 8
 
   // version settings
   sbt-version-override         : "0.13.9"
@@ -377,7 +377,7 @@ build += {
   }
 
   //
-  // Typesafe-developed projects: degvelopment branches / master / unstable releases
+  // Typesafe-developed projects: development branches / master / unstable releases
   //
   ${vars.base} {
     name: "async"
@@ -388,6 +388,21 @@ build += {
   ${vars.base} {
     name: "slick"
     uri:  "https://github.com/"${vars.slick-ref}
+    // without this dbuild doesn't pick up that one of the subprojects has this dependency.
+    // it doesn't even make sense; it seems to me that testNGSettings should not be adding
+    // a dependency of the plugin to the libraryDependencies of the test code.
+    // the line in question is:
+    //   https://github.com/sbt/sbt-testng-interface/blob/ca730f705f48af2139f39bc726b474afec072738/plugin/src/main/scala/de/johoop/testngplugin/TestNGPlugin.scala#L44
+    // I think it's a confusion of levels, but maybe I'm missing something. - ST 8/27/15
+    deps.inject: ["de.johoop#sbt-testng-interface"]
+    // disable fragile tests (https://github.com/scala/community-builds/issues/12#issuecomment-149941055)
+    extra.exclude: ["osgitests"]
+  }
+
+  ${vars.base} {
+    name: "sbt-testng-interface",
+    uri: "https://github.com/"${vars.sbt-testng-interface-ref}
+    extra.projects: ["sbt-testng-interface"]  // just the interface, we don't need to build the plugin
   }
 
   ${vars.base} {

--- a/common.conf
+++ b/common.conf
@@ -126,7 +126,7 @@ vars: {
   sbt-republish-ref            : "typesafehub/sbt-republish.git"
   scalaz-ref                   : "scalaz/scalaz.git#series/7.1.x"
   discipline-ref               : "typelevel/discipline.git#v0.2"
-  scala-js-ref                 : "scala-js/scala-js.git#v0.6.4"
+  scala-js-ref                 : "scala-js/scala-js.git"
 
   // version settings
   sbt-version-override         : "0.13.9"

--- a/common.conf
+++ b/common.conf
@@ -148,7 +148,7 @@ options.resolvers: {
   01: "local"
   02: "maven-cache: https://scala-ci.typesafe.com/artifactory/central/"
   03: "typesafe-ivy-releases: https://repo.typesafe.com/typesafe/ivy-releases/, [organisation]/[module]/[revision]/[type]s/[artifact](-[classifier]).[ext], bootOnly"
-  04: "sbt-plugin-releases: https://dl.bintray.com/sbt/sbt-plugin-releases/"${vars.ivyPat}
+  04: "sbt-plugin-releases: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/"${vars.ivyPat}
   09: "dbuild-snapshots: http://repo.typesafe.com/typesafe/temp-distributed-build-snapshots"${vars.ivyPat}
   10: "maven-central"
   11: "old-typesafe-maven-releases: https://repo.typesafe.com/typesafe/maven-releases" // for play 2.3.9's netty-http-pipelining 1.1.2

